src/zx/FunctionalityConstruction.cpp

```diff
@@ -30,23 +32,23 @@ namespace zx {
 
     void FunctionalityConstruction::addZSpider(ZXDiagram& diag, zx::Qubit qubit,
                                                std::vector<Vertex>& qubits,
-                                               const Expression& phase, EdgeType type) {
-        auto newVertex = diag.addVertex(
+                                               const PiExpression& phase, EdgeType type) {
+        auto new_vertex = diag.addVertex(
                 qubit, diag.getVData(qubits[qubit]).value().col + 1, phase,
                 VertexType::Z);
 
-        diag.addEdge(qubits[qubit], newVertex, type);
-        qubits[qubit] = newVertex;
+        diag.addEdge(qubits[qubit], new_vertex, type);
+        qubits[qubit] = new_vertex;
     }
 
     void FunctionalityConstruction::addXSpider(ZXDiagram& diag, Qubit qubit,
                                                std::vector<Vertex>& qubits,
-                                               const Expression& phase, EdgeType type) {
-        const auto newVertex = diag.addVertex(
+                                               const PiExpression& phase, EdgeType type) {
```

Comment:
```suggestion
                                               const PiExpression& phase, const EdgeType type) {
```

Reply:
Running out of funny const replies....

Reply:
At some point, there will be no more `const`s to add. Until then, you better think of some good excuses ;o)